repository_service_tuf_worker/services/storage/awss3.py

```diff
@@ -0,0 +1,155 @@
+# SPDX-FileCopyrightText: 2022-2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+from io import BytesIO
+from typing import List, Optional
+
+import awswrangler
+import boto3
+from botocore.exceptions import ClientError
+from securesystemslib.exceptions import StorageError  # noqa
+from tuf.api.metadata import Metadata, T, Timestamp
+from tuf.api.serialization import DeserializationError
+
+from repository_service_tuf_worker import parse_if_secret
+from repository_service_tuf_worker.interfaces import IStorage, ServiceSettings
+
+
+class AWSS3(IStorage):
+    def __init__(
+        self,
+        bucket: str,
+        access_key: str,
+        secret_key: str,
+        region: Optional[str] = None,
+        endpoint_url: Optional[str] = None,
+    ) -> None:
+        self._bucket: Optional[str] = bucket
```

Comment:
Bucket is not optional:
```suggestion
        self._bucket: str = bucket
```